src/types/DomainDefinitions.ts

```diff
@@ -31,21 +31,23 @@ export enum PreconditionType {
   Role = "role",
 }
 
+export interface IFieldDefinition {
+  fieldType: string;
+  label: string;
+  required?: boolean;
+  minLength?: number;
+  maxLength?: number;
+  pattern?: string;
+  minValue?: number;
+  maxValue?: number;
+  minDate?: Date;
+  maxDate?: Date;
+}
 export interface IRoleDefinitionText {
   roleType: string;
   roleName: string;
-  fields: {
-    fieldType: string;
-    label: string;
-    required?: boolean;
-    minLength?: number;
-    maxLength?: number;
-    pattern?: string;
-    minValue?: number;
-    maxValue?: number;
-    minDate?: Date;
-    maxDate?: Date;
-  }[];
+  fields: IFieldDefinition[];
+  issuerFields?: IFieldDefinition[];
```

Comment:
IMO, it would have been better to not have made this field optional actually. In my opinion, `DomainReader` should handle whether or not this field is available from the on-chain role definition and if it isn't available, then set the property to an empty array. This is so that client wouldn't have perform an undefined check. https://stackoverflow.com/a/332248
But I think it's not a big deal 😄